core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2771,6 +2780,84 @@ private CompiledQuery vacuum(SqlExecutionContext executionContext) throws SqlExc
+        if (Chars.equalsLowerCaseAscii(tok, "prepare")) {
+
+            if (lockedReaders.size() > 0) {
+                throw SqlException.position(lexer.lastTokenPosition()).put("Another snapshot command in progress");
+            }
+
+            path.of(configuration.getSnapshotRoot()).slash();
+            configuration.getSnapshotDirTimestampFormat().format(
+                    configuration.getMicrosecondClock().getTicks(),
+                    configuration.getDefaultDateLocale(),
+                    null,
+                    path);
+
+            int snapshotLen = path.length();
+            if (ff.exists(path.slash$())){
+                throw CairoException.instance(0).put("Snapshot dir already exists [dir=").put(path).put(']');
+            }
+            path.trimTo(snapshotLen);
+            try (TableListRecordCursorFactory factory = new TableListRecordCursorFactory(configuration.getFilesFacade(),
+                    configuration.getRoot())) {
+                try (RecordCursor cursor = factory.getCursor(executionContext)) {
+                    final Record record = cursor.getRecord();
+                    try {
+                        while (cursor.hasNext()) {
+                            CharSequence tableName = record.getStr(0);
+                            TableReader reader = getTableReaderForStatement(executionContext, tableName, "snapshot");
+                            lockedReaders.add(reader);
+
+                            path.concat(configuration.getDbDirectory()).concat(tableName).slash$();
+
+                            if (ff.mkdirs(path, configuration.getMkDirMode()) != 0) {
+                                throw CairoException.instance(ff.errno()).put("Could not create [dir=").put(path).put(']');
+                            }
+
+                            int rootLen = path.length();
+                            try (MemoryCMARW mem = Vm.getCMARWInstance()) {
+                                mem.smallFile(ff, path.trimTo(rootLen).concat(TableUtils.META_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
+                                reader.getMetadata().dumpTo(mem);
+                                mem.close(false);
+                                mem.smallFile(ff, path.trimTo(rootLen).concat(TableUtils.TXN_FILE_NAME).$(), MemoryTag.MMAP_DEFAULT);
+                                reader.getTxFile().dumpTo(mem);
+                                mem.close(false);
+                            }
+
+                            path.trimTo(snapshotLen);
+                        }
+                        ff.sync(); // commit filesystem caches to disk
+                    } catch (Throwable e) {
+                        path.trimTo(snapshotLen);
+                        path.$();
+                        ff.rmdir(path);
+                        Misc.freeObjList(lockedReaders);
+                        lockedReaders.clear();
+                        throw e;
```

Comment:
I'd log it too, that's unexpected error

Reply:
Addressed in 286c96a452b25b8ab536c0c243f97201bcc90b96